### ISSUE TITLE
Braille: Add support for the HIMS Braille display Braille Emotion

### ISSUE THREAD

**Issue by Nael-Sayegh** [nvaccess/nvda#16842]
### Is your feature request related to a problem? Please describe.
With the new HIMS Braille display "Braille Emotion," quite a few keys are not recognized or many keyboard shortcuts are not defined.

### Describe the solution you'd like
Add keyboard shortcuts to this display, fully support it with these specific keys, and accept the simultaneous pressing of certain keys.

### Describe alternatives you've considered
None

**Comment by LeonarddeR** [comment]
cc @EdKweon

**Comment by EdKweon** [comment]
@Nael-Sayegh 
Could you please specify exactly which keys are not being recognized? Additionally, it would be helpful if you could separate your feedback for HID and Serial. Please note that while single keys have been defined, shortcut combination keys have not been defined separately. It seems that this is an issue that needs to be discussed and added according to the specific requirements of NVDA users. If you have any particular direction in mind, please let us know. Thank you.

**Comment by Nael-Sayegh** [comment]
@EdKweon Hello,
I didn't quite understand your message. What do you mean by HID and serial? 

Also, I don't know if you are using this Braille display, but how can I indicate the buttons to define since they don't have names?

And I didn't understand the end of your message.

**Comment by EdKweon** [comment]
@Nael-Sayegh 
I have a Braille eMotion.

Braille eMotion can be connected to NVDA using three methods: HID, Serial, and Bluetooth. When you first entered the Connectivity(ScreenReader) app to connect to NVDA, did you select the USB connection?

`Quite a few keys are not recognize`
I want to know what keys are not recognized. Give me a  example that not works.

Thanks.

**Comment by Nael-Sayegh** [comment]
@EdKweon When I connected the braille display, I connected it via Bluetooth.
The keys not recognized on the display are W, B, and C.
The Home button on the front of the display has an assigned function but does not have a name to be remapped.
The F3 and F4 keys on the display do not work with other control keys like F1 or F8.
The arrow keys on the right and left sides of the display cannot be remapped to simulate the down arrow and up arrow, for example.

**Comment by EdKweon** [comment]
@Nael-Sayegh 
I can't understand  what is W, B and C. Does it mean dot keys? (ex: 'W' is dot 2 - 4 - 5 -6)
As far as I know, the braille eMotion device only has F1, F2, F3, and F4 keys. Are you perhaps referring to the Cursor Keys when you mention Arrow Keys?

**Comment by Nael-Sayegh** [comment]
@EdKweon I'm talking about the three buttons at the top left of the device, where 'w', 'b', and 'C' are written in Braille on each of them, in that order."

If you require a more technical approach, especially if this is for instructions or documentation, you could say:

"I am referring to the three buttons located at the top left corner of the device. Each button has a Braille label: 'w' on the first button, 'b' on the second, and 'C' on the third, in that specific order.

**Comment by EdKweon** [comment]
@Nael-Sayegh 
Thanks. You mean Wifi, Blutooth, Connectivity buttons. 

Let me confirm my understanding. Are you saying that currently, the cursor key and home key on the display are not mapped in the Bluetooth connection?(Maybe other keys are more)

Secondly, what actions do you want the three buttons at the top left to perform in NVDA? As far as I know, these keys do not have any assigned functions in control keys.

**Comment by Nael-Sayegh** [comment]
> * Par exemple, la touche Accueil sur l'écran ne fonctionne pas -> f1 ~ f8 (touches de contrôle) ne fonctionnent pas.
>
@EdKweon I didn't understand your request.

**Comment by EdKweon** [comment]
@Nael-Sayegh 
I was literally asking if the home key or cursor keys on the display are not working in Bluetooth connection.

**Comment by Nael-Sayegh** [comment]
@EdKweon The cursors work correctly in Bluetooth.
What do you mean by the home button?
If it's the one between points 1 and 4, when I press it, it shows me something like "clipboard content," but I've never managed to understand what it does.
If you're talking about the middle button on the front, it works by having the F1 function.

**Comment by EdKweon** [comment]
@Nael-Sayegh 
It works normally. Then I will return to the main point. I am sorry, but the three buttons on the top left of the Braille eMotion will not be supported in the NVDA screen reader key dictionary. This is also true for USB connections. Secondly, is the home key on the display performing the F1 function when connected via Bluetooth? As far as I remember, it should perform the ESC function. What function does the F2 key on display perform?

**Comment by Nael-Sayegh** [comment]
@EdKweon Hello,

Sorry for my late reply.
Even in bluetooth mode, the home button executes the F1 key.
And the F2 key on the display executes tab.
It is the F1 key that executes escape.

**Comment by Nael-Sayegh** [comment]
@EdKweon I find it unfortunate that this product has so few commands in NVDA, requires so many adjustments, and has so many non-reassignable keys.

**Comment by EdKweon** [comment]
@Nael-Sayegh 
It seems that the basic key functions of eMotion are working well. It appears that you're asking for NVDA key mapping for Braille eMotion. Currently, it should be possible to operate NVDA with the common HIMS combination keys for eMotion. Could you test if dot1+dot3+dot6+space works as CapsLock and dot1+dot3+dot5+dot6+space works as Alt+F4?

**Comment by Nael-Sayegh** [comment]
@EdKweon dot1+dot3+dot6 emulates the NVDA key and dot1+dot3+dot5+dot6+space correctly performs alt+F4.
Apart from the lack of documentation on the shortcuts for the display, my problem is primarily the inability to create shortcuts with certain key combinations, the impossibility of remapping the keys on each side of the display as well as the front keys at each end and the home key.
If all these problems can be resolved, it could already make using the display easier.

**Comment by EdKweon** [comment]
@Nael-Sayegh 
Okay. I understand what you saying. I will check the problem till next week.
Thanks!

**Comment by Nael-Sayegh** [comment]
@EdKweon Hello,
Have you been able to find any solutions regarding this issue?

**Comment by ABuffEr** [comment]
Hi @EdKweon and all,
is there a possibility to have cursors routing working in HID mode?

**Comment by ABuffEr** [comment]
Hi,
sorry, I just upgraded the firmware, and cursors routing work now! 🎉

**Comment by seanbudd** [comment]
@ABuffEr - does this mean this issue can be closed?

**Comment by ABuffEr** [comment]
Hi @seanbudd,
not directly, from what I understand: situation is better now, but it seems that for HID and Bluetooth profiles are missing various commands regard to the old HIMS profile. But discussion is not so well structured, and I have the Braille eMotion just from a day, so asking to open a new more precise issue could be a good proposal imho.

**Comment by seanbudd** [comment]
I agree - I'm going to close this. I think bug reports should be opened for anything remaining
